hashring/hashring.go

```diff
@@ -41,16 +42,17 @@ type HashRingConfiguration struct {
 	ReplicaPoints int
 }
 
+// HashRing stores strings on a consistent hash ring. HashRing internally uses
+// a Red-Black Tree to achieve O(log N) lookup and insertion time.
 type HashRing struct {
-	hashfunc      func([]byte) uint32
+	sync.RWMutex
+
+	hashfunc      func(string) int
 	replicaPoints int
 
-	servers struct {
-		byAddress map[string]bool
-		tree      *RedBlackTree
-		checksum  uint32
-		sync.RWMutex
-	}
+	servers  map[string]struct{}
```

Comment:
I had some questions about using a `slice` or a `map` to implement a set, but gladly we have people on the internet that had those questions first! http://blog.dubbelboer.com/2015/08/15/slice-vs-map.html

a `map` seems to be the right choice here.